### PR TITLE
Add `FVPostprocessorDirichletBC`

### LINKED ISSUE #16477: `FVPostprocessorDirichletBC`
## Reason
A FV (finite volume) equivalent of `PostprocessorDirichletBC` is needed. One example use case is for a vessel inlet temperature condition in a fluid flow sim where the vessel exterior is handled in a separate MultiApp which will vary the the inlet temperature value.

## Design
This should behave as a normal `FVDirichletBC` except the value should be dynamically set by a postprocessor.

## Impact
This is a new feature.

### PR DESCRIPTION
**Pull request by smharper**
Closes #16477.

This PR adds the eponymous BC. It also introduces the new base class `FVDirichletBCBase`. This base class is needed because FV flux kernels need to be able to specially treat Dirichlet BCs, and a base class allows for all sorts of Dirichlet BCs to be handled generically. (Note that in the develop branch there is an alternate implementation of this polymorphism where `FVFunctionDirichletBC`s inherit from `FVDirichletBC`s, but they have to remove the `value` entry from the list of parameters which is unsatisfying.)

### CODE REVIEW

File: framework/include/fvbcs/FVDirichletBCBase.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Base class for finite volume Dirichlet boundaray conditions. A base class is
+ * needed because Dirichlet BCs must be handled specially when computing fluxes
+ * on faces in the FV method.
+ */
+class FVDirichletBCBase : public FVBoundaryCondition
+{
+public:
+  /**
+   * Class constructor.
```
Comment (rwcarlsen) [inline framework/include/fvbcs/FVDirichletBCBase.h:23]: comments on constructors that say "constructor" are... well I don't like them.

File: framework/include/fvbcs/FVPostprocessorDirichletBC.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+#pragma once
+
+#include "FVDirichletBCBase.h"
+
+class FVPostprocessorDirichletBC : public FVDirichletBCBase
+{
+public:
+  /**
+   * Class constructor.
```
Comment (rwcarlsen) [inline framework/include/fvbcs/FVPostprocessorDirichletBC.h:18]: yuck

File: framework/include/fvbcs/FVDirichletBC.h [hunk 3]
```diff
@@ -9,12 +9,12 @@
...
+#include "FVDirichletBCBase.h"
 
 /**
  * Base class for
  */
-class FVDirichletBC : public FVBoundaryCondition
+class FVDirichletBC : public FVDirichletBCBase
 {
 public:
   /**
```
Comment (rwcarlsen) [inline framework/include/fvbcs/FVDirichletBC.h:20]: Github won't let me post a review comment on text below the diffs here - but throw in an `override` on the boundaryValue func for good measure.

**Review by rwcarlsen — CHANGES_REQUESTED** [review]
Just a few very minor things.

File: framework/include/fvbcs/FVDirichletBCBase.h (hunk 1 above)
Comment (smharper) [reply framework/include/fvbcs/FVDirichletBCBase.h:23]: Agreed. But honestly I don't think a constructor should require documentation unless it does something unexpected. Can I totally remove this doxygen block or do our code standards require that we have one?

Comment (rwcarlsen) [reply framework/include/fvbcs/FVDirichletBCBase.h:23]: I'm okay with you killing it.

**Review by rwcarlsen — APPROVED** [review]

**Pull request merged**